feat: Add option to skip session selection in ballot application

Tailmon opened the pull request:
This pull request introduces the ability to skip session selection during the ballot process by adding a new configuration option and updating the relevant functions to handle this new option.

### Configuration Changes:
* [`config.toml`](diffhunk://#diff-28043ff911f28a5cb5742f7638363546311225a63eabc365af5356c70d4deb77R19-R21): Added a new configuration option `SkipSessionSelection` to allow skipping the session selection step.

### Code Changes:
* `main.py`: 
  * Updated the `apply_for_single_session` function to accept a `skip_session_selection` parameter and modified the logic to skip session selection if this parameter is set.
  * Modified the `start_single_ballot_process` function to retrieve the `SkipSessionSelection` configuration and use it to control session selection logic. [[1]](diffhunk://#diff-b10564ab7d2c520cdd0243874879fb0a782862c3c902ab535faabe57d5a505e1R337-R341) [[2]](diffhunk://#diff-b10564ab7d2c520cdd0243874879fb0a782862c3c902ab535faabe57d5a505e1R350-R357) [[3]](diffhunk://#diff-b10564ab7d2c520cdd0243874879fb0a782862c3c902ab535faabe57d5a505e1L365-R372)

Tailmon committed:
feat: Add option to skip session selection in ballot application (#1)

eyalmazuz commented:
Seems like I can't understand this PR fully or the rationality behind it

Can you please provide a specific use-case or detailed explanation to what it does?

Tailmon commented:
Yeah, I made these changes specifically for applying to Hasu 4th Live.

For events like this, the code only works for a specific day, and the application URL does not have a “Session Selection” page — it goes directly to the page where you enter your code.

With that in mind, I adjusted the logic to support this use case. You can now configure it via the `config.toml` file to skip the session selection step when it’s not present.

eyalmazuz commented:
> For events like this, the code only works for a specific day, and the application URL does not have a “Session Selection” page — it goes directly to the page where you enter your code.

Yeah I now understand better the use case, it actually makes sense. 

> With that in mind, I adjusted the logic to support this use case. You can now configure it via the config.toml file to skip the session selection step when it’s not present.

Even though eplus doesn't have too many options and complex logic, I think having more options to the toml might be too bloated in the future, some options can either be handled via implicit declarations.

Instead of having another variable to control if to skip the day/session-selection landing page. I would personally prefer that the logic of the landing page and serial input would be decoupled into two different methods and then we could just check if a landing page exists.

Actually, this part of the code technically already does that
```py
sessions = driver.find_elements(By.XPATH, "//div[@class='page-content']//a")
sessions_name = {session.text[1:-5] for session in sessions}
```

so we could reuse this logic instead of having another method to do similar checks

so in theory the code would look like:

```py
driver.get(entry_url)
sessions = driver.find_elements(By.XPATH, "//div[@class='page-content']//a")
sessions_name = {session.text[1:-5] for session in sessions}
while codes:
    current_code = codes.pop()
    driver.get(entry_url)
    if sessions_name:
    # if the set isn't empty then you found any days/sessions so it's the landing page
        for session_name in sessions_name:
            select_day_or_session(driver)
            result = enter_code(driver)
            rest_of_ballot_process(driver)
    else:
    # if the set **is** empty then you can't find any days/sessions so it's the serial page
         result = enter_code(driver)
         rest_of_ballot_process(driver)
```


I only slightly worried about 
```py
sessions = driver.find_elements(By.XPATH, "//div[@class='page-content']//a")
sessions_name = {session.text[1:-5] for session in sessions}
```

returning false-positives from other HTML elements that might exist in the serial page so maybe need to change the ``find_elements`` logic to something better

Tailmon committed:
fix: Update button text for session application and login, increase wait time for ballot options

Tailmon committed:
refactor: auto-detect session selection instead of config flag

Remove SkipSessionSelection config option. Instead, check if session
links exist on the page to determine whether to show session selection.
This avoids config bloat and handles the use case implicitly.

eyalmazuz commented:
LGTM
